include/fmt/format.h

```diff
@@ -1788,12 +1807,32 @@ auto write_float(OutputIt out, const DecimalFP& fp,
 
 template <typename Char, typename OutputIt, typename T,
           FMT_ENABLE_IF(std::is_floating_point<T>::value)>
-auto write(OutputIt out, T value, basic_format_specs<Char> specs,
-           locale_ref loc = {}) -> OutputIt {
+FMT_CONSTEXPR20 auto write(OutputIt out, T value,
```

Comment:
With `FMT_CONSTEXPR20` defined to `constexpr`, line 1842 is no longer portable and causes compile failures when `std::isfinite()` isn't marked as `constexpr`. The C++ standard doesn't mandate this function to be `constexpr`. Other functions from `cmath` are probably affected as well.

I suggest introducing portable, `constexpr`-capable replacements.

Reply:
fixed, somehow I faced with std::signbit, but didn't face with isfinite and isinf thinking 🤔